### ISSUE TITLE
Shortcode within Rich Text element

### ISSUE THREAD

**Issue by infinitnet** [soflyy/oxygen-bugs-and-features#515]
It would make sense to support shortcode within Rich Text elements like other page builders do. This would make things a lot easier to insert widgets or shortcode that's required within a line of text.

**Comment by davidjbullock** [comment]
+1 for this.  Most other builders allow this behavior, and it's an expected behavior with new users that they can use shortcodes in text blocks.

Allowing this behavior is in keeping with the principle of least surprise for the user.

**Comment by yankiara** [comment]
This is definitely needed! And you were kind to set it as enhancement ;-)
Shortcode is a major core functionnality in Wordpress, there's no reason it wouldn't work in basic elements like rich text!

**Comment by AThoss** [comment]
This is needed.

**Comment by flore89** [comment]
+1 please.

**Comment by triphitz** [comment]
Yes please, surprised this hasn't been added by now. +1

**Comment by zuroma999** [comment]
This is sorely needed. Even for the regular text element it'd be great, just like other page builders. Now, to get shortcode output next to text we have to wrap text and shortcode elements in extra divs, which is so counter to Oxygen's normally clean and simple html.

**Comment by joeyhead** [comment]
How is this not a thing yet? I moved away from Thrive Architect which is way less customizable than Oxygen but I could add shortcodes directly in texts with it. Now I'm having a hard time just to display shortcodes that returns text values next to simple texts. Please make this happen. This is needed. +1

**Comment by triphitz** [comment]
> How is this not a thing yet? I moved away from Thrive Architect which is way less customizable than Oxygen but I could add shortcodes directly in texts with it. Now I'm having a hard time just to display shortcodes that returns text values next to simple texts. Please make this happen. This is needed. +1

Agreed! How can we get the team to focus on implementing this feature? This one should have been done months ago!

**Comment by MasterOfWeb** [comment]
+1
and while you at it, please supprt nested shortcode or dynamic content in shortcodes. something like [my_shortcode text="[oxygen data='title' ]" ]

**Comment by HmCody** [comment]
This just makes so much sense to include. Please make it happen and soon!

**Comment by foaad1961** [comment]
I totally agreed with all who are asking for the shortcode within the rich text element. I had to go and add the Advanced Editor Tools plugin (Tiny Mce) to add other editing tools.

**Comment by foaad1961** [comment]
I have even found solutions for it, but I assume the technical team at Soflyy/Oxygen Builder team can manage it much easier than I can do:
https://www.smashingmagazine.com/2012/05/wordpress-shortcodes-complete-guide/

**Comment by erfdroid** [comment]
I was surprised this is not working. While googling a solution I got here. Therefore, +1

**Comment by violllettta** [comment]
In my [pre-Oxygen] sidebars and pages I use "Display Posts" shortcode which allows for lists of posts with lots of parameters and "Shortcode Menu" shortcode which calls any of the Wordpress menus.

**Comment by d-packs** [comment]
please add this feature.

**Comment by JeusuDigital** [comment]
Agreed this should be added as it makes it awkward doing inline short codes like for instance in my case I have a short code setup for the antispambot function and a lot of the time I need to wrap it around emails that are in-between text. 

For anyone looking for a workaround you can use a Code Block and echo the short code out like this:

`<?php echo do_shortcode('[your_shortcode_here]')` ?>

**Comment by d-packs** [comment]
> For anyone looking for a workaround you can use a Code Block and echo the short code out like this:
> 
> `<?php echo do_shortcode('[your_shortcode_here]')` ?>

That's not really practical when you are trying to embed shortcodes in a block of text.

**Comment by JeusuDigital** [comment]
@d-packs It is currently the only way to do it, you have to add the text to the code block like this:

`<p>My block of text <?php echo do_shortcode('[your_shortcode_here]') ?> with a shortcode</p>`

**Comment by gdewey** [comment]
+1  please add!!   on any dynamic text content oxy shortcodes work but not third party  ones

**Comment by GESign** [comment]
+1 please add!!! This would be important. Thank you so much! ;)

**Comment by stevenrabulan** [comment]
+1. There are parts of my site where I'd like to put in dynamic snippets of text that I update from time to time (e.g. expiration dates/event dates/prices that might change, etc)

**Comment by Cenay** [comment]
This is exactly my use case. I have 4 different types of "camps" that have start and end dates, weekly prices, daily prices, and a rich text list of the various "weeks" available that appear 5 or 6 times on a long sales page. I would LOVE to be able to add ACF fields for the start date, end date, weekly price, daily price and the rich text that contains the unordered list of weeks (and their status as sold out or not) and attach to each camp page. However, I can't pull in the text area added in ACF to the rich text component. 

Anyone know of a work around?

**Comment by zuroma999** [comment]
> This is exactly my use case. I have 4 different types of "camps" that have start and end dates, weekly prices...
> 
> Anyone know of a work around?

Only solution I know of is code. How I handle it: I put a Rich Text Element inside a wrapped shortcode. The RTF element content has placeholders, for example:

"Here is some text and here is the event title [~get_event_title~] and it starts at [~get_event_date~]"

and the wrapping shortcode calls my shortcode function which grabs all the [~x~] text, does the shortcode, replaces the text, and then returns the final html.

**Comment by Cenay** [comment]
So, it turns out putting the HTML into a textarea type field in ACF, will actually render correctly! So now, I put the things that change from year to year and camp to camp into ACF fields, and render them via dynamic text, inlcuding the unordered list in the ACF field! Works great!

**Comment by Nate0ED** [comment]
Yes please implement this - the shortcode wrapper is causing CLS issues for a membership website as it dynamically loads based on user roles. I had to use code blocks to render shortcode inline with text and it's all really clunky. Every other builder I've used in the past worked with inline shortcode.

**Comment by realmalikov** [comment]
+1 Shortcodes are such an essential part of WordPress that they should work in every text field. In my opinion, there is no need for a separate shortcode element. The shortcode wrapper, on the other hand, is useful.
